src/org/zaproxy/zap/extension/frontendscanner/ExtensionFrontEndScanner.java

```diff
@@ -229,25 +226,18 @@ public boolean onHttpResponseReceive(HttpMessage msg) {
 
                     int historyReferenceId = msg.getHistoryRef().getHistoryId();
 
-                    // 67000 is a lower estimate, made by counting the characters in
-                    //   * files/frontendscanner/front-end-scanner.js
-                    //   * files/scripts/scripts/client-side-passive/scan-jwt-token.js
-                    StringBuilder injectedContentBuilder = new StringBuilder(67000)
-                        .append("<script type='text/javascript'>")
-                        .append("var frontEndScanner=(function() {")
-                        .append("const HISTORY_REFERENCE_ID = " + historyReferenceId + ";")
-                        .append("const CALLBACK_ENDPOINT = '" + frontEndApiUrl + "';");
+                    StringBuilder injectedContentBuilder = new StringBuilder(65000)
+                        .append("<script src='" + frontEndApiUrl

```

Comment:
when using a StringBuilder you should always append each individual string (as before) rather than concatenating the strings, otherwise you loose all the benefit.